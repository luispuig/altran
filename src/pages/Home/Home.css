._Home .city-appear {
	opacity: 0.01;
}

._Home .city-appear.city-appear-active {
	opacity: 1;
	transition: opacity 900ms ease-in;
	-webkit-transition-delay: 1200ms; /* Safari */
	transition-delay: 1200ms !important;
}

._Home .city-appear.city-appear-active:nth-of-type(1) {
	-webkit-transition-delay: 0ms; /* Safari */
	transition-delay: 0ms !important;
}

._Home .city-appear.city-appear-active:nth-of-type(2) {
	-webkit-transition-delay: 300ms; /* Safari */
	transition-delay: 300ms !important;
}

._Home .city-appear.city-appear-active:nth-of-type(3) {
	-webkit-transition-delay: 600ms; /* Safari */
	transition-delay: 600ms !important;
}

._Home .city-appear.city-appear-active:nth-of-type(4) {
	-webkit-transition-delay: 900ms; /* Safari */
	transition-delay: 900ms !important;
}
