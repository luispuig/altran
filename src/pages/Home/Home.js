// @flow
import React from 'react'
import City from './components/City';

type _Props = {};

const  temperatures = [
  {id:3936456,  temperature:15.45,  date: new Date('2017-12-06T10:57:16.462Z')},
  {id:3448439,  temperature:22.06,  date: new Date('2017-12-06T10:57:16.449Z')},
  {id:3871336,  temperature:12,     date:new Date('2017-12-06T10:57:16.438Z')}
];


class Home extends React.Component<_Props> {
  render() {
    return (
        <div className="container row">
          <div className="col">
            <City id={1} name="City Name" state="request" temperatures={[]}/>
          </div>
          <div className="col">
            <City id={1} name="City Name" state="success" temperature={25} temperatures={temperatures}/>
          </div>
          <div className="col">
            <City id={1} name="City Name" state="fail"  temperatures={[]}/>
          </div>
          <div className="col">
            <City id={1} name="City Name" state="fail" error="error message" temperatures={temperatures}/>
          </div>
        </div>
    );
  }
}

export default Home
