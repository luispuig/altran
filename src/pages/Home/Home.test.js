import React from 'react';
import ReactDOM from 'react-dom';

import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';

import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
configure({ adapter: new Adapter() });


import Home from './Home';


const initialData = { temperature: 0, state: 'request', error: undefined  };
const testState = {
  cities: [
    { id: 3871336, name: "Santiago",      ...initialData },
    { id: 3435910, name: "Buenos Aires",  ...initialData },
    { id: 3936456, name: "Lima",          ...initialData },
    { id: 3448439, name: "Sao Paulo",     ...initialData }
  ],
  cities_temperatures: [
    { id:3871336, date: new Date('2017-12-06T14:14:58.080Z'), temperature: 25 },
    { id:3435910, date: new Date('2017-12-07T14:14:58.080Z'), temperature: 26 },
    { id:3936456, date: new Date('2017-12-08T14:14:58.080Z'), temperature: 27 },
    { id:3448439, date: new Date('2017-12-09T14:14:58.080Z'), temperature: 28 },
  ]
};

const store = createMockStore(testState);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><Home /></Provider>, div);
});

it('match on Shallow mode', () => {
  const component = mount(<Provider store={store}><Home /></Provider>, store);
  const tree = shallowToJson(component, {mode: 'shalow'});
  expect(tree).toMatchSnapshot();
});


it('should start fetching', () => {
  jest.useFakeTimers();

  const cityUpdateAll_Mock = jest.fn();

  const component = mount(<Provider store={store}><Home cityUpdateAll={cityUpdateAll_Mock} /></Provider>, store);

  expect(cityUpdateAll_Mock.mock.calls.length).toBe(1); // Start Fetching con ComponentDidMount();

  // Run timer
  jest.runOnlyPendingTimers();
  expect(cityUpdateAll_Mock.mock.calls.length).toBe(2); // Fetching on tic timer();

  // unmount component
  component.unmount();

  // Run timer
  jest.runOnlyPendingTimers();
  expect(cityUpdateAll_Mock.mock.calls.length).toBe(2); // NO Fetching on tic timer();

  jest.clearAllTimers();
});
