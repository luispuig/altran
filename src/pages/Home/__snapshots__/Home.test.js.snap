// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match on Shallow mode 1`] = `
<div
  className="container row"
>
  <div
    className="col"
  >
    <City
      id={1}
      name="City Name"
      state="request"
      temperatures={Array []}
    />
  </div>
  <div
    className="col"
  >
    <City
      id={1}
      name="City Name"
      state="success"
      temperature={25}
      temperatures={
        Array [
          Object {
            "date": 2017-12-06T10:57:16.462Z,
            "id": 3936456,
            "temperature": 15.45,
          },
          Object {
            "date": 2017-12-06T10:57:16.449Z,
            "id": 3448439,
            "temperature": 22.06,
          },
          Object {
            "date": 2017-12-06T10:57:16.438Z,
            "id": 3871336,
            "temperature": 12,
          },
        ]
      }
    />
  </div>
  <div
    className="col"
  >
    <City
      id={1}
      name="City Name"
      state="fail"
      temperatures={Array []}
    />
  </div>
  <div
    className="col"
  >
    <City
      error="error message"
      id={1}
      name="City Name"
      state="fail"
      temperatures={
        Array [
          Object {
            "date": 2017-12-06T10:57:16.462Z,
            "id": 3936456,
            "temperature": 15.45,
          },
          Object {
            "date": 2017-12-06T10:57:16.449Z,
            "id": 3448439,
            "temperature": 22.06,
          },
          Object {
            "date": 2017-12-06T10:57:16.438Z,
            "id": 3871336,
            "temperature": 12,
          },
        ]
      }
    />
  </div>
</div>
`;
